### ISSUE TITLE
getdefaultlocale() deprecated in Python 3.11

### ISSUE THREAD

**Issue by pesekon2** [OSGeo/grass#2538]
**Describe the bug**

Initializing GRASS with Python 3.11, I am getting the following warning messages:

```
/grass/bin.x86_64-pc-linux-gnu/grass:69: DeprecationWarning: Use setlocale(), getencoding() and getlocale() instead
  ENCODING = locale.getdefaultlocale()[1]
/grass/bin.x86_64-pc-linux-gnu/grass:1421: DeprecationWarning: Use setlocale(), getencoding() and getlocale() instead
  language, encoding = locale.getdefaultlocale()
```

Not annoying so much, but it could cause us some problems once the function will be fully removed from `locale`. 

**Possible solution**

Change to `getlocale()`. Will dive deeper to find out if it can cause some problems.

**Comment by pesekon2** [comment]
See a discussion touching this topic [here](https://github.com/python/cpython/issues/90817)

**Comment by neteler** [comment]
This is meanwhile a blocker in Fedora 37:
https://bugzilla.redhat.com/show_bug.cgi?id=2126608

**Comment by neteler** [comment]
>  it might be easier to accept the fixes (#2550 and #2539) and then see if somebody reports any issue on Windows?

Yes, I agree. Will you merge it?
